### PR TITLE
[system-probe] Fix fl6 offset guessing when not running in root net ns

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?

Fix an issue where we fail to guess flowi6 offsets when the system-probe is not running in the root network namespace and the root network namespace (i.e. the host) has ipv6 enabled. We run the code for acquiring an ipv6 link local network address in the root network namespace to get around this.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/tracer/offsetguess.go [hunk 1]
```diff
@@ -796,19 +803,33 @@ func newEventGenerator(ipv6 bool) (*eventGenerator, error) {
...
 		return nil, nil
 	}
 
-	linkLocal, err := getIPv6LinkLocalAddress()
-	if err != nil {
-		// TODO: Find a offset guessing method that doesn't need an available IPv6 interface
-		log.Debugf("unable to find ipv6 device for udp6 flow offset guessing. unconnected udp6 flows won't be traced: %s", err)
-		return nil, nil
-	}
+	// it is necessary to run the following code in
```
Comment (akarpz) [inline pkg/network/tracer/offsetguess.go:811]: out of curiosity do you know why the host & root NS would have IPv6 enabled but the container NS would not?

Comment (hmahmood) [reply pkg/network/tracer/offsetguess.go:811]: I think you have to explicitly enable it. With `docker` it is not enabled by default.

File: pkg/network/tracer/offsetguess.go [hunk 2]
```diff
@@ -796,19 +803,33 @@ func newEventGenerator(ipv6 bool) (*eventGenerator, error) {
...
+	// the root network namespace, as the network
+	// namespace where we're running may not have
+	// ipv6 enabled. we are here because the host
+	// does have ipv6 enabled (ipv6 bool is true).
+	// if system-probe's network namespace does
+	// not have ipv6 enabled, the below code will fail
+	// since none of the interfaces will have an
+	// ipv6 link local address assigned
```
Comment (heyronhay) [inline pkg/network/tracer/offsetguess.go:819]: Is this part of the comment saying "if system-probe is running in a container, and that's container's network namespace doesn't have ipv6 enabled, then this won't work"?

**Review by heyronhay — APPROVED** [review]
Looks good - just one question for my understanding!

File: pkg/network/tracer/offsetguess.go (hunk 2 above)
Comment (hmahmood) [reply pkg/network/tracer/offsetguess.go:819]: Essentially, yes.

**Review by akarpz — APPROVED** [review]

**Pull request merged**